network/src/lib.rs

```diff
@@ -25,15 +25,15 @@ pub type MessageResult = Result<tonic::Response<types::Empty>, anyhow::Error>;
 
 #[derive(Debug)]
 #[must_use]
-pub struct CancelHandler<T>(pub tokio::task::JoinHandle<T>);
+pub struct CancelOnDropHandler<T>(tokio::task::JoinHandle<T>);
```

Comment:
❤️

Reply:
Since you are touching this, could you also please add a quick comment about the behaviour as well?